.github/workflows/pipeline.yml

```diff
@@ -354,12 +358,20 @@ jobs:
       # Check the deploy technology (containers and s3 should be covered differently) - conditionals cover only cache/artifact
       - name: Restore the build from cache
         uses: actions/cache/restore@v3
-        if: ${{ inputs.use_artifact == 'false' }}
+        if: ${{ inputs.use_artifact == 'false' && inputs.build_cache == '' }}
         id: restore-build-cache
         with:
           path: ${{ env.OUTPUT_DIST_PATH }}
           key: ${{ runner.os }}-build-${{ steps.commit.outputs.short }}
 
+      - name: Restore the previous build from cache
+        uses: actions/cache/restore@v3
+        if: ${{ inputs.use_artifact == 'false' && inputs.build_cache != '' }}
+        id: restore-build-cache
+        with:
+          path: ${{ env.OUTPUT_DIST_PATH }}
+          key: ${{ runner.os }}-build-${{ inputs.build_cache }}
```

Comment:
@DarkoKukovec @danipavic , should we change the ${{ runner.os }} to ${{ inputs.environment }} so we have `production-build-{short_hash}` instead of `Linux-build-{short_hash}`?

Reply:
Changed it just to keep environments separated as much as possible.